### PR TITLE
Add bs1770 3 fir

### PR DESCRIPTION
**Pull request by colekas-ssim**
Add interpolation filter that was originally used in [libebur128](https://github.com/jiixyj/libebur128/blob/master/ebur128/ebur128.c) to support BS.1770-3 measurements with FIR interpolation filter.

### CODE REVIEW

File: libavfilter/f_ebur128.c [hunk 1]
```diff
@@ -496,6 +518,59 @@ static int config_audio_output(AVFilterLink *outlink)
             !ebur128->true_peaks_per_frame || !ebur128->swr_ctx)
             return AVERROR(ENOMEM);
 
+        if (ebur128->peak_mode & FILTERED) {
```
Comment (michelle-zhang-cw) [inline libavfilter/f_ebur128.c:522]: && FILTERED ?

Comment (colekas-imax) [reply libavfilter/f_ebur128.c:522]: I'm checking to see if the `FILTERED` bit is set

**Review by michelle-zhang-ssimwave — APPROVED** [review]

File: libavfilter/f_ebur128.c [hunk 2]
```diff
@@ -73,12 +75,29 @@ struct integrator {
     struct hist_entry *histogram;   ///< histogram of the powers, used to compute LRA and I
 };
 
+typedef struct interp_filter {
+  unsigned int count;  /* Number of coefficients in this subfilter */
+  unsigned int* index; /* Delay index of corresponding filter coeff */
+  double* coeff;       /* List of subfilter coefficients */
+}interp_filter;
```
Comment (tcrugnale-imax) [inline libavfilter/f_ebur128.c]: Nitpick: Space between `}` and `interp_filter`

File: libavfilter/f_ebur128.c [hunk 3]
```diff
@@ -176,6 +197,7 @@ static const AVOption ebur128_options[] = {
         { "none",   "disable any peak mode",   0, AV_OPT_TYPE_CONST, {.i64 = PEAK_MODE_NONE},          INT_MIN, INT_MAX, A|F, "mode" },
         { "sample", "enable peak-sample mode", 0, AV_OPT_TYPE_CONST, {.i64 = PEAK_MODE_SAMPLES_PEAKS}, INT_MIN, INT_MAX, A|F, "mode" },
         { "true",   "enable true-peak mode",   0, AV_OPT_TYPE_CONST, {.i64 = PEAK_MODE_TRUE_PEAKS},    INT_MIN, INT_MAX, A|F, "mode" },
+        { "tpfiltered",   "enable true-peak mode filtered for BS.1770-3",   0, AV_OPT_TYPE_CONST, {.i64 = PEAK_MODE_TRUE_PEAKS_FILTERED},    INT_MIN, INT_MAX, A|F, "mode" },
```
Comment (tcrugnale-imax) [inline libavfilter/f_ebur128.c]: Are these also filtered for BS.1770-4?  Should we be clear about that by referring to "BS.1770-3+"?

File: libavfilter/f_ebur128.c [hunk 4]
```diff
@@ -637,17 +712,50 @@ static int filter_frame(AVFilterLink *inlink, AVFrame *insamples)
         const double *swr_samples = ebur128->swr_buf;
         int ret = swr_convert(ebur128->swr_ctx, (uint8_t**)&ebur128->swr_buf, 19200,
                               (const uint8_t **)insamples->data, nb_samples);
+        double acc = 0;
```
Comment (tcrugnale-imax) [inline libavfilter/f_ebur128.c]: These should be moved to the scope of the if statement on line 725

**Review by tomcrugnale-ssimwave — APPROVED** [review]
LGTM, only a couple of minor things.

**Pull request merged**